src/main/java/com/amazon/opendistroforelasticsearch/sql/query/DefaultQueryAction.java

```diff
@@ -152,6 +152,9 @@ public void setFields(List<Field> fields) throws SqlParseException {
                     MethodField method = (MethodField) field;
                     if (method.getName().toLowerCase().equals("script")) {
                         handleScriptField(method);
+                        if (method.getExpression() instanceof SQLCastExpr) {
```

Comment:
I just checked other functions and this issue (return all fields) seems common for default DSL format. Could you add IT for JDBC format to verify if it works? For other functions, JDBC format looks good.